R/topics.R

```diff
@@ -167,3 +167,59 @@ topic_must <- function(message, topic) {
     x = paste0("Not ", encodeString(topic, quote = "'"))
   ))
 }
+
+content_info <- function(content_entry, index, pkg, section) {
+
+  if (!grepl(".*::.*", content_entry)) {
+    topics <- pkg$topics[select_topics(content_entry, pkg$topics),]
+    tibble::tibble(
+      path = topics$file_out,
+      aliases = purrr::map2(
+        topics$funs,
+        topics$name,
+        ~ if (length(.x) > 0) .x else .y
+      ),
+      name = list(topics$name),
+      title = topics$title,
+      icon = find_icons(topics$alias, path(pkg$src_path, "icons"))
+    )
+  } else { # topic from another package
+    names <- strsplit(content_entry, "::")[[1]]
+    if (!rlang::is_installed(names[1])) {
+      abort(
+        sprintf(
+          "%s must be installed if it is mentioned in the reference index.",
+          names[1]
+        )
+      )
+    }
+
+    path <- downlit::href_topic(names[2], names[1])
+    if (is.na(path)) {
+      abort(
+        sprintf(
+          "Could not find an href for topic %s of package %s",
+          names[2], names[1]
+        )
+      )
+    }
+
+    rd_name <- fs::path_ext_set(fs::path_file(path), "Rd")
+    # adapted from printr
+    db <- tools::Rd_db(names[1])
+    Rd <- db[[rd_name]]
+    # adapted from printr
+    # https://github.com/yihui/printr/blob/0267c36f49e92bd99e5434f695f80b417d14e090/R/help.R#L69
+    sections <- sub('^\\\\', '', unlist(lapply(Rd, attr, 'Rd_tag')))
+    title <- as.character(Rd[sections == "title"][[1]])
+
+    tibble::tibble(
+      path = path,
+      aliases = sprintf("%s (from %s)", names[2], names[1]),
```

Comment:
Should it get a `()` after the function name? (Do we know if it's it a function or not?)

Reply:
No we don't e.g. both `sass::font_face` (rd name) and `sass::font_google` (function name) work. :grimacing:

Reply:
I've tweaked the code so that one could write `rlang::is_installed()` and get "`is_installed()` (from rlang)" in the reference index but now I'm seeing that if linking to a whole topic one might actually want to list everything in that topic.